src/ape_cache/query.py

```diff
@@ -302,9 +306,13 @@ def update_cache(self, query: QueryType, result: Iterator[BaseInterfaceModel]):
         clause = self.cache_update_clause(query)
         if str(clause):
             logger.debug(f"Caching query: {query}")
-            with self.database_connection as conn:
-                conn.execute(
-                    clause.values(self.get_cache_data(query, result)).prefix_with(  # type: ignore
-                        "OR IGNORE"
+            try:
+                with self.database_connection as conn:
+                    conn.execute(
+                        clause.values(self.get_cache_data(query, result)).prefix_with(  # type: ignore
+                            "OR IGNORE"
+                        )
                     )
-                )
+
+            except QueryEngineError as err:
+                logger.error(f"Database not initiated: {err}")
```

Comment:
Should probably skip caching entirely if the database file isn't available (no need to log anything in that scenario), and if there are any further issues then the database is probably corrupted and the user should probably purge and re-init it or something
```suggestion
                logger.warning(f"Database corruption: {err}")
```

Reply:
[8b2a092](https://github.com/ApeWorX/ape/pull/680/commits/8b2a09266ec5834c9c8ebd74747f0ad21d8f5e70)